include/swift/AST/DiagnosticsSema.def

```diff
@@ -2197,6 +2197,11 @@ NOTE(overridden_here,none,
      "overridden declaration is here", ())
 NOTE(overridden_here_can_be_objc,none,
      "add '@objc' to make this declaration overridable", ())
+// SWIFT_ENABLE_TENSORFLOW
+ERROR(override_missing_differentiable,none,
+      "%0 is a missing a `@differentiable` definition that is required in the base protocol",
```

Comment:
It would be nice to include the exact attribute the declaration is missing. [Here](https://github.com/apple/swift/blob/2ce6f9c40acc4713189fa0a4999ac04ee6f786f8/lib/Sema/TypeCheckProtocol.cpp#L2173) is how we do that when a conforming type member is missing a `@differentiable` attribute. Could you do something similar (and, if possible, reuse the diagnostics there)?

Reply:
I implemented this and will push soon, but where should I put `computeDerivativeGenericEnvironment` so we don't have duplicate code?

Reply:
For now, I pushed a version that duplicates it but can move it once you tell me where it's best to put it.

Reply:
Thanks! I wonder if `computeDerivativeGenericEnvironment` is necessary for the protocol case because `where` clauses won't exist in `@differentiable` attributes in protocol requirements. You should be able to just use the function (the decl being checked)'s own generic environment.

Reply:
It's necessary I believe in order to check if the `wrt` parameters are inferable. Say you had this:
```swift
protocol Test {
  @differentiable(where Scalar: Differentiable)
  func test(value: Scalar)
}

protocol Test2: Test where Scalar Differentiable {
  func test(value: Scalar)
}
```
In this case you want the error message to omit the `wrt` parameters because they are inferable and I believe `computeDerivativeGenericEnvironment` is necessary to do that. Is this right? I might also be misunderstanding something.

Reply:
@rxwei I verified this is the case and added a test for this. I also made a minor fix in the code that prints `@differentiable` attributes.

Reply:
@rxwei I removed `computeDerivativeGenericEnvironment` and so this should be ready now.

Reply:
Great!